core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/TextScanSourceOpExec.scala

```diff
@@ -0,0 +1,37 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import edu.uci.ics.texera.workflow.common.operators.source.SourceOperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.Schema
+
+import java.io.{BufferedReader, FileReader}
+import scala.jdk.CollectionConverters.asScalaIteratorConverter
+
+class TextScanSourceOpExec private[text] (
+    val desc: TextScanSourceOpDesc,
+    val startOffset: Int,
+    val endOffset: Int,
+    val outputAsSingleTuple: Boolean
+) extends SourceOperatorExecutor {
+  private var schema: Schema = _
+  private var reader: BufferedReader = _
+  private var rows: Iterator[String] = _
+
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+    if (outputAsSingleTuple) {
+      Iterator(Tuple.newBuilder(schema).add(schema.getAttribute("line"), rows.mkString).build())
```

Comment:
here when in `outputAsSingleTuple` mode, you rejoin those rows into a string. however, did you add the `\n` line separator back?

Reply:
depending on the file format/encoding, spliting a file into rows then rejoin them together may corrupt the file content. Can we remove this split + rejoin logic, and simply "read all content" and wrap it as a STRING field?

Reply:
@Yicong-Huang For `outputAsSingleTuple` mode, would we still want to allow users to specify an offset and max number of lines then?  Would this also potentially corrupt the file since we will need to extract a subset of the lines?

Reply:
As per our discussion, the operator in `outputAsSingleTuple` will directly read the entire file into the tuple in this case without offsets or limits and without splitting the file into lines, as to avoid corrupting the file.